### PR TITLE
Listening history - Display search bar

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This adds a search bar on the listening history screen.

## Testing Instructions

1. Login with an account with sufficient listening history
2. Go to `Profile` tab -> `Listening History`
3. ✅ Notice that a search bar now exists at the top
4. Long press an episode
5. ✅ Notice that multi-select bar is displayed at the top 
6. Dismiss the multi-select toolbar
7. Scroll the listening history list 
8. ✅ Notice that search bar scrolls behind the toolbar
9. Go to `Profile` tab -> `Starred`
10. ✅ Notice that search bar is not present on the screen
11. Go to `Profile` tab -> `Downloads`
12. ✅ Notice that search bar is not present on the screen
13. Disable feature flag
14. Go to `Profile` tab -> `Listening History`
15. ✅ Notice that search bar is not present on the screen

## Screenshots or Screencast 

https://github.com/user-attachments/assets/dcf4824d-91aa-49aa-acbf-98727ce1add7

**Themed**

**Landscape**

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews (the only component added is a search bar which has a preview available)
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by CookieyedCodes** [comment]
It would be great if you could search the podcast name & the date, it would limit the search to those feeds and how many podcasts were listened too on a spefic date 🤔

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/ProfileEpisodeListViewModel.kt [hunk 1]
```diff
@@ -23,17 +29,34 @@ class ProfileEpisodeListViewModel @Inject constructor(
...
         val episodeListFlowable = when (mode) {
             is ProfileEpisodeListFragment.Mode.Downloaded -> episodeManager.observeDownloadEpisodes()
             is ProfileEpisodeListFragment.Mode.Starred -> episodeManager.observeStarredEpisodes()
             is ProfileEpisodeListFragment.Mode.History -> episodeManager.observePlaybackHistoryEpisodes()
         }
-
-        episodeList = episodeListFlowable.toLiveData()
+        viewModelScope.launch {
+            episodeListFlowable.asFlow()
+                .stateIn(viewModelScope)
```
Comment (MiSikora) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/ProfileEpisodeListViewModel.kt:46]: I `stateIn()` needed for something?

**Review by MiSikora — APPROVED** [review]

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/ProfileEpisodeListViewModel.kt (hunk 1 above)
Comment (ashiagr) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/ProfileEpisodeListViewModel.kt]: I don't think so. Removed in 70a48f1.

**Pull request merged**